File: UglyTrivia/WithTests/C#/GameRunner.cs
using System;

namespace UglyTrivia
{
  public class GameRunner
  {

    public static void PlayTheGame()
    {
      Run(new Random());
    }

    private static bool notAWinner;

    public static void Run(Random rand)
    {
      Game aGame = new Game();

      Console.WriteLine("\n\n--- Trivia Game ---");

      aGame.add("Chet");
      aGame.add("Pat");
      aGame.add("Sue");

      do
      {
        aGame.roll(rand.Next(5) + 1);

        if (rand.Next(9) == 7)
        {
          notAWinner = aGame.wrongAnswer();
        }
        else
        {
          notAWinner = aGame.wasCorrectlyAnswered();
        }

      } while (notAWinner);
    }
  }
}

File: UglyTrivia/WithTests/Java/GameRunner.java

import java.util.Random;


public class GameRunner {

        public static void playTheGame()
        {
            playTheGame(new Random());
        }

	public static void playTheGame(Random rand) {

                boolean notAWinner;
		Game aGame = new Game();

                System.out.println("\n\n--- Trivia Game ---");
		
		aGame.add("Chet");
		aGame.add("Pat");
		aGame.add("Sue");
		
		do {
			
			aGame.roll(rand.nextInt(5) + 1);
			
			if (rand.nextInt(9) == 7) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.wasCorrectlyAnswered();
			}	
			
		} while (notAWinner);	
	}
}
